Build error on Ubuntu 18.04 and GCC 7.3.0

I get the below error when building on 18.04 which has gcc 7.3.0. The error goes away when I add `-Wno-unused-parmater` to the main `CMakeLists.txt`. 
```
/home/vaivaswatha/c/zilliqa/build/src/libMessage/Message.pb.cc: In member function 'virtual google::protobuf::uint8* ProtoMessage::Nested::InternalSerializeWithCachedSizesToArray(bool, google::protobuf::uint8*) const':
/home/vaivaswatha/c/zilliqa/build/src/libMessage/Message.pb.cc:317:10: error: unused parameter 'deterministic' [-Werror=unused-parameter]
     bool deterministic, ::google::protobuf::uint8* target) const {
```

--- a/src/libMessage/CMakeLists.txt
+++ b/src/libMessage/CMakeLists.txt
@@ -1,4 +1,5 @@
 protobuf_generate_cpp(PROTO_SRC PROTO_HEADER Message.proto)
 add_library (Message SHARED ${PROTO_HEADER} ${PROTO_SRC})
+target_compile_options(Message PRIVATE "-Wno-unused-parameter")
 target_include_directories (Message PUBLIC ${PROJECT_SOURCE_DIR}/src)
 target_link_libraries (Message PUBLIC ${PROTOBUF_LIBRARY})

